src/Generator/Base/Property.cs

```diff
@@ -32,21 +30,13 @@ internal abstract class Property : Writable
 
     internal string Type { get => type; set => HandleTypeSetter(value); }
 
-    internal string? NonInterfaceType { get; set; }
-
     internal string Name { get => name; set => name = CapitalizeFirstLetter(value); }
 
     internal string JsonName { get; set; } = string.Empty;
 
-    internal string? DictionaryPatchType { get; set; }
-
-    internal bool NeedsConvertMethod { get; set; }
-
-    internal bool UseNonInterfaceType => NonInterfaceType != Type;
-
     internal bool JsonIgnore { get; set; } = false;
 
-    internal bool Nullable { get; set; } = false;
+    internal bool Nullable { get; } = false;
```

Comment:
This looks odd. If it's always not nullable then we probably don't need this?

Maybe we look at this later.